python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -821,6 +823,22 @@ def resolve_isPartitioned(self, _graphene_info: ResolveInfo) -> bool:
     def resolve_isObservable(self, _graphene_info: ResolveInfo) -> bool:
         return self._external_asset_node.is_observable
 
+    def resolve_isExecutable(self, _graphene_info: ResolveInfo) -> bool:
+        from dagster._core.definitions.asset_spec import (
+            SYSTEM_METADATA_KEY_ASSET_VARIETAL,
+            AssetVarietal,
+        )
+
+        metadata_value = self._external_asset_node.metadata.get(SYSTEM_METADATA_KEY_ASSET_VARIETAL)
+        if not metadata_value:
+            varietal_text = None
+        else:
+            check.inst(metadata_value, TextMetadataValue)  # for guaranteed runtime error
+            assert isinstance(metadata_value, TextMetadataValue)  # for type checker
+            varietal_text = metadata_value.value
+
+        return AssetVarietal.is_executable(varietal_text)
```

Comment:
would be good to move this down on to `ExternalAssetNode` and have a test against a checked in serialized copy to protect this moving forward

Reply:
will do this in followup